### DISCUSSION TITLE
Terminology

### DISCUSSION THREAD

**Discussion by johnbent** [tekinged/missing#375] [Announcements]
To help us better discuss tekinged.com, the following terms are important to know:
1. **Database**
  a. A system for storing information. Almost all of the information that comprises tekinged.com is stored in a database.
  b. Conceptually, it is reasonable to think of a database as a set of tables and each table is similar to a spreadsheet table with column names such as 'palauan' and 'english'. Then there are a set of rows with values within each column (e.g. _BLAI_ and house)
3. **Web pages**
  a. These are the actual pages that appear within the tekinged.com website (example: [Palauan Example Text](https://tekinged.com/misc/alltext.php)
  b. Most of the web pages contain simple programming instructions that retrieve information from the database, and transform it into the web page
2. **Database entry**
  a. A individual row in a database table.
  b. All of the tekinged.com words and definitions are stored in a database table. 
  c. Some of the key column names in this table are 'pal', 'eng', 'tags', 'id', 'root', and 'pos'. 
    - 'pos' is for part of speech. 
    - 'tags' is how we build many of the "List ..." pages such as '[Word List: Animals'](https://tekinged.com/show_words.php?lookup=animal). 
    - 'id' is a unique number for each database entry that allows precise and unambiguous access to that database entry. We need 'id' instead of just using 'pal' because there are a few words in Palauan that appear multiple times in this table. For example, there are multiple database entries for the Palauan word _MECHAS_:
      - noun; old woman; titled woman; foreign woman; male's father's sisters; girlfriend; wife.
      - ergative verb; get blackened with soot or ink; (pot) get burned or discolored.                                           
      - noun; coconut at later stage (between medecheduch and metau) when shell blackens and husk turns yellowish brown. 
      - state verb; having the qualities of an old woman.
    - 'root' will be described below
2. **Word group**
  a. A word group is a set of one or more words that are logically grouped together. For example, _BLAI_ and _BLIL_ are in one word group, and _MLAI_ and _MLIL_ are in a second word group.
  b. In the main search functionality, the tekinged.com search page (almost always) shows words grouped into their word groups.
3. **Root word**
  a. Within a word group, one of the words is considered to be the 'root word'. For example, _BLAI_ is the root word in its word group. 
  b. In the database, we mark root words by setting their 'root' value equal to their 'id' value. For example, both the 'id' and 'root' for _BLAI_ are the number 3256. This means that the root word for _BLAI_ is itself.
5. **Stem word**
  a. A stem word is a word that is not a root word and that is linked to a root word and therefore is part of the word group with the root word. For example, _BLIL_ is a stem word that is linked to _BLAI_ (by setting its root also to 3256) and therefore appears in the _BLAI_ word group.
6. **GitHub**
  a. GitHub is an online platform that the tekinged.com Committee uses to discuss our work.
7. **GitHub Issues**
  a. The committee creates a GitHub Issue for discussing possible additions, deletions, and changes to the database table
  b. Within each Issue, we discuss it and attempt to find convergence. We then attempt to use the [AUTOPARSE feature](https://github.com/tekinged/missing/discussions/348) to simplify making the database changes. 
  c. We can also use Issues to report bugs and request new features in the web page 
8. **GitHub Discussions**
  a. For other discussions within the Committee (for example this one), we can use a GitHub Discussion.

**Comment by ChrisPerrette** [comment]
Thank you, this will help communication greatly.
